quartz/plugins/emitters/cname.ts

```diff
@@ -0,0 +1,28 @@
+import { FilePath, joinSegments } from "../../util/path"
+import { QuartzEmitterPlugin } from "../types"
+import fs from "fs"
+
+export function extractDomainFromBaseUrl(baseUrl: string) {
+  const url = new URL(`https://${baseUrl}`)
+  return url.hostname
+}
+
+export const CNAME: QuartzEmitterPlugin = () => ({
+  name: "CNAME",
+  getQuartzComponents() {
+    return []
+  },
+  async emit({ argv, cfg }, _content, _resources, _emit): Promise<FilePath[]> {
+    if (!cfg.configuration.baseUrl) {
+      console.warn("No baseUrl set while CNAME emitter enabled.")
```

Comment:
```suggestion
      console.warn(chalk.yellow("CNAME emitter requires \`baseUrl\` to be set in your configuration"))
```

^ also import `chalk` here, all of our warnings should be noticeably different (e.g. https://github.com/jackyzha0/quartz/blob/8fc7b9f4c68bb8c41a156e92cfbddf3144678dcc/quartz/plugins/emitters/contentPage.tsx#L61)

Reply:
Done